controller/snapshot_controller.go

```diff
@@ -468,13 +469,20 @@ func (sc *SnapshotController) handleAttachmentTicketCreation(snap *longhorn.Snap
 
 	existingVA := va.DeepCopy()
 	defer func() {
-		if err != nil {
-			return
-		}
 		if reflect.DeepEqual(existingVA.Spec, va.Spec) {
 			return
 		}
 
+		if checkIfSnapshotExists {
+			// It is possible we are reconciling a cached version of a deleted snapshot (only if deletionTimestamp is
+			// set). Confirm that the snapshot still exists on the API server before creating an attachment ticket to
+			// prevent an orphan attachment (see longhorn/longhorn#6652). Avoid checking until here to limit
+			// requests to the API server.
+			if _, err = sc.ds.GetLonghornSnapshotUncached(snap.Name); err != nil {
+				return // Either the snapshot no longer exists or we failed to check.
```

Comment:
Can you clarify whether you are asking for me to:

1. Only log and then NOT return the error, or
2. Log AND return the error?

The contents of the error are already logged at the top level of the reconciliation when the error is returned.

For 1, we rely on the error to stop us from continuing to reconcile. I could further modify the function signature to return both a boolean (indicating success or failure) and an error if desired. Then, return "false, nil" in the case we get a "not found".

For 2, is it redundant to do so, since we will see the error contents twice in quick succession?

(I think you could argue that this is an error in the same way as "the object has been modified" is. It is expected to see that pretty frequently in our logs.)

Reply:
It's just about logging any error  eventually instead of implicitly digesting it.

However, we are fine as per your explanation below.

> The contents of the error are already logged at the top level of the reconciliation when the error is returned.